docs/installing.rst

```diff
@@ -27,21 +28,54 @@ In your ``conf.py`` file:
     :ref:`supported-dependencies`
         Officially Supported versions of Python, Sphinx, and other dependencies.
 
+
 .. note::
 
    Adding this theme as an extension is what enables localization of theme
    strings in your translated output. If these strings are not translated in
    your output, either we lack the localized strings for your locale, or you
    are using an old version of the theme.
 
+   ..
+      comment about this note: it's possibly not necessary to add the theme as an extension.
+      Rather, this is an issue caused by setting html_theme_path.
+      See: https://github.com/readthedocs/readthedocs.org/pull/9654
+
+
+.. _howto_upgrade:
+
+How to upgrade
+--------------
+
+.. tip:: 
+    **Read the Docs users:** To know which version of ``sphinx-rtd-theme`` you are running, you can always check your latest build outputs. One or more steps in the documentation build will install python packages, one of which is ``sphinx-rtd-theme``.
+
+Adding ``sphinx-rtd-theme`` to your project's ``requirements.txt`` will make pip install the latest compatible version of the theme.
+
+If you want to test a **pre-release**, you need to be explicit about the version in ``requirements.txt``.
+Otherwise, pip will ignore pre-releases. Add for instance ``sphinx-rtd-theme==1.1.0b3`` to test a pre-release.
+
+.. tip::
+    We always recommend that you pin the version of Sphinx that your project is built with.
+    We'll make sure not to release sphinx-rtd-theme that installs together with a Sphinx theme that it is incompatible with.
+    
+    It can never be guaranteed that sphinx-rtd-theme is compatible with everything in your project.
+    If your project is complex or needs reproducible documentation builds, you should pin or put an upper bound on ``sphinx-rtd-theme``.
+    For instance, you could add this to ``requirements.txt``::
+    
+        # Ensure that we do not automatically include changes from the next major series
+        sphinx-rtd-theme>=1,<2
```

Comment:
We have some great documentation here already that would be better to point to:
https://docs.readthedocs.io/en/stable/guides/reproducible-builds.html

Reply:
Ah yes, I also ref'ed that in the announcement: https://github.com/readthedocs/blog/pull/184

I'll add a reference :+1:

Reply:
Well, I meant to say that this content should be replaced with a link to the page we have authored. This is generic Python dependency management advice, so no specific to the theme, and pur guide on reproducible builds is thorough.

Most important, this is slightly at odds with what we recommend in the reproducible builds guide and we should probably just communicate to users one, single vision for reproducible builds.

Reply:
Is there already a consensus about treating the theme as all other dependencies?

The advice was intended as being specific to the theme... I would take into greater consideration to advice about the effects of exact pinning vs. loose pinning. The benefits of a loose pinning or no pinning at all are greater be. I was thinking that it's up to the reader to decide and take responsibility for what level of "reproducibility" they want.

Reply:
Yup, we have consensus here. Our guidance is that guide that I linked, and is that users should always explicitly pin every dependency.

This is partly self-serving, as reproducible builds don't inflict support time when some random, backwards incompatible, transitive dependency puts out a release.

Users can do whatever they want to, but we prefer to communicate a _singular_ opinion here: always explicitly pin every one of your dependencies.

Reply:
Sounds good

From experience, I disagree - but it's a good thing. I have benefited from not having to bump versions of the theme or keep an eye out for updates. I just always get the latest version in my builds, no errors or issues or efforts made.

Reply:
In addition to the self-serving part.. asking people to declare versions is also asking them to be aware of new releases from the beginning. So that can plant a seed for more engagement with theme releases.

If we're gonna tackle this in the future, it could be by having a second section called "Non-reproducible builds". A thought for now.

On the self-serving part: We also don't benefit from having issues that are fixed in later theme releases and then having to instruct to people to upgrade.

Anyways, blah blah blah, will replace it with a link to the generic chapter :+1:

Reply:
Yeah I get that. The middle ground between free for all and using a lock file is covered in that guide, using pip-tools, and seems like a great approach so far.

You can still upgrade easily to the latest sphinx_rtd_theme based on your specifiers. But it's not an automatic process. Either way, every version you're installing is explicitly defined in a requirements file, so there is no guessing which version you're installing.

Reply:
This is fixed now :+1: